be/src/vec/data_types/serde/data_type_datetimev2_serde.cpp

```diff
@@ -112,13 +114,23 @@ void DataTypeDateTimeV2SerDe::write_column_to_arrow(const IColumn& column, const
                     binary_cast<UInt64, DateV2Value<DateTimeV2ValueType>>(col_data[i]);
             datetime_val.unix_timestamp(&timestamp, ctz);
 
-            if (scale > 3) {
+            if (scale > 0) {
                 uint32_t microsecond = datetime_val.microsecond();
-                timestamp = (timestamp * 1000000) + microsecond;
-            } else if (scale > 0) {
-                uint32_t millisecond = datetime_val.microsecond() / 1000;
-                timestamp = (timestamp * 1000) + millisecond;
+
+                if (scale > 3) {
+                    timestamp = (timestamp * 1000000) + microsecond;
+                } else {
+                    uint32_t millisecond = microsecond / 1000;
+                    timestamp = (timestamp * 1000) + millisecond;
+                }
+            } else {
+                timestamp *= 1000000;
             }
+            timestamp -= std::chrono::duration_cast<std::chrono::seconds>(
```

Comment:
sub `1970` timestamp does not seem to be the root cause,
it seems to be a problem caused by incorrect `DATETIME` `scale`.

Reply:
> sub `1970` timestamp does not seem to be the root cause, it seems to be a problem caused by incorrect `DATETIME` `scale`.

I've revert the changes of timestmp scale.